src/api/search_transactions.rs

```diff
@@ -0,0 +1,254 @@
+};
+use serde_json::json;
+use sqlx::FromRow;
+
+use crate::{
+  operation, util::DEFAULT_TOKEN_ID, ChainStatus, MinaMesh, MinaMeshError, OperationType, TransactionStatus,
+  UserCommandType,
+};
+
+#[derive(Debug, FromRow)]
+pub struct UserCommand {
+  pub id: Option<i32>,
+  pub command_type: UserCommandType,
+  pub fee_payer_id: Option<i32>,
+  pub source_id: Option<i32>,
+  pub receiver_id: Option<i32>,
+  pub nonce: i64,
+  pub amount: Option<String>,
+  pub fee: Option<String>,
+  pub valid_until: Option<i64>,
+  pub memo: Option<String>,
+  pub hash: String,
+  pub block_id: Option<i32>,
+  pub sequence_no: Option<i32>,
+  pub status: TransactionStatus,
+  pub failure_reason: Option<String>,
+  pub state_hash: Option<String>,
+  pub chain_status: Option<ChainStatus>,
+  pub height: Option<i64>,
+  pub total_count: Option<i64>,
+  pub fee_payer: String,
+  pub source: String,
+  pub receiver: String,
+  pub creation_fee: Option<String>,
+}
+
+impl UserCommand {
+  pub fn decoded_memo(&self) -> Option<String> {
+    let memo = self.memo.clone().unwrap_or_default();
+    match bs58::decode(memo).into_vec() {
+      Ok(decoded_bytes) => {
+        let cleaned = &decoded_bytes[3 .. decoded_bytes[2] as usize + 3];
+        Some(String::from_utf8_lossy(cleaned).to_string())
+      }
+      Err(_) => None,
+    }
+  }
+
+  pub fn into_block_transaction(self) -> BlockTransaction {
+    let decoded_memo = self.decoded_memo().unwrap_or_default();
+    let amt = self.amount.clone().unwrap_or_else(|| "0".to_string());
+
+    let mut operations = Vec::new();
+    let mut operation_index = 0;
+
+    // Operation 1: Fee Payment
+    operations.push(operation(
+      operation_index,
+      Some(&format!("-{}", self.fee.unwrap_or_else(|| "0".to_string()))),
+      AccountIdentifier {
+        address: self.fee_payer.clone(),
+        metadata: Some(json!({ "token_id": DEFAULT_TOKEN_ID })),
+        sub_account: None,
+      },
+      OperationType::FeePayment,
+      Some(&self.status),
+      None,
+      None,
+    ));
+
+    operation_index += 1;
+
+    // Operation 2: Account Creation Fee (if applicable)
+    if let Some(creation_fee) = &self.creation_fee {
+      if let Ok(fee_value) = creation_fee.parse::<i64>() {
+        if fee_value > 0 {
+          operations.push(operation(
+            operation_index,
+            Some(&format!("-{}", creation_fee)),
+            AccountIdentifier {
+              address: self.receiver.clone(),
+              metadata: Some(json!({ "token_id": DEFAULT_TOKEN_ID })),
+              sub_account: None,
+            },
+            OperationType::AccountCreationFeeViaPayment,
+            Some(&self.status),
+            None,
+            None,
+          ));
+
+          operation_index += 1;
+        }
+      }
+    }
+
+    // Decide on the type of operation based on command type
+    match self.command_type {
+      UserCommandType::Payment => {
+        // Operation 3: Payment Source Decrement
```

Comment:
Stylistic preference: can we place comments above the match arm?

Reply:
Done in 739e291282337ee4af26adf41304637dc802ea9e.